error when compiling with gcc 12.2.0

<!-- markdownlint-disable MD041 -->

Trying to compile a branch different only by a few lines form master on supermuc NG.
(https://github.com/SeisSol/SeisSol/pull/905/files)

By mistake I compile with gcc 12.2 and got the following error:
```
di73yeq4@login04:~> spack install -j 40 seissol@mw2 convergence_order=4 dr_quad_rule=dunavant equations=elastic precision=single ^easi +python
[+] /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/asagi/1.0.1-gcc-12.2.0-mqgdhwx
[+] /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/nghttp2/1.57.0-oneapi-2023.1.0-rekqxl2
[+] /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/zlib-ng/2.1.4-oneapi-2023.1.0-tbqz43r
[+] /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/ncurses/6.4-oneapi-2023.1.0-zlsh5oo
[+] /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/impalajit/main-gcc-12.2.0-p6zl5is
[+] /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/lua/5.3.2-gcc-12.2.0-xr2pugk
==> python@3.10.10 : has external module in ['python/3.10.10-extended']
[+] /lrz/sys/spack/release/23.1.0/views/python/3.10.10-extended (external python-3.10.10-k7f2wyb2g45imofbrdtqekaijmq6hclb)
[+] /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/yaml-cpp/0.6.2-gcc-12.2.0-ak2irdf
[+] /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/eigen/3.4.0-gcc-lbmyl47
[+] /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/gmake/4.4.1-gcc-s2cwoxq
[+] /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/intel-oneapi-mpi/2021.9.0-gcc-xizuusf
[+] /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/libaec/1.0.6-gcc-3e6k7yf
[+] /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/pkgconf/1.8.0-gcc-u3gkc4x
[+] /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/zlib/1.2.13-gcc-p5ywc53
[+] /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/libxsmm/1.17-gcc-2dvcdwk
[+] /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/numactl/2.0.14-gcc-zaa36r4
[+] /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/metis/5.1.0-gcc-kougmmh
[+] /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/openssl/3.1.3-oneapi-2023.1.0-yr23atf
[+] /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/py-pspamm/develop-gcc-12.2.0-lq7lvdi
[+] /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/py-setuptools/68.0.0-gcc-12.2.0-h2renn2
[+] /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/easi/1.3.0-gcc-12.2.0-to3va2m
[+] /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/parallel-netcdf/1.11.2-gcc-o2g36gf
[+] /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/hdf5/1.10.9-gcc-hbsptk3
[+] /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/memkind/1.13.0-gcc-xjtus63
[+] /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/parmetis/4.0.3-gcc-nypuwzn
[+] /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/curl/8.4.0-oneapi-2023.1.0-eub6cvj
[+] /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/netcdf-c/4.6.3-gcc-aq23kth
[+] /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/cmake/3.27.7-gcc-12.2.0-fxbprdj
==> Installing seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl [29/29]
==> No binary for seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl found: installing from source
==> No patches needed for seissol
==> seissol: Executing phase: 'cmake'
==> seissol: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j40'

4 errors found in build log:
     473    /dss/lrzsys/sys/spack/user/release/23.1.0/skylake_avx512/spack/lib/spack/env/gcc/g++ -DALIGNED_REAL_SIZE=4 -DALIGNMENT=64 -DCONVERGENCE_ORDER=4 -DEIGEN_NO_CUDA -DH5_BUILT_AS_DYNAMIC_LIB -DLOGLEVEL0=2 -DLOGLEVEL="" -DLOG_LEVEL=2 -DNUMBER_OF
            _RELAXATION_MECHANISMS=0 -DOMP -DOMPI_SKIP_MPICXX -DPARALLEL -DREAL_SIZE=4 -DUSE_ASAGI -DUSE_ELASTIC -DUSE_HDF -DUSE_METIS -DUSE_MPI -DUSE_NETCDF -DUSE_NUMA_AWARE_PINNING -DUSE_PARMETIS -DUSE_PLASTICITY_NB -DVECTORSIZE=64 -D_FILE_OFFSET_BI
            TS=64 -D_GNU_SOURCE -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -D_POSIX_C_SOURCE=200809L -Ddunavant -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-build-r32ac7j/src/generat
            ed_code -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/src/Equations/elastic -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33
            ftnxsiyueorce3uobl/spack-src/src/Initializer/BatchRecorders -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/intel-oneapi-mpi/2021.9.0-gcc-xizuusf/mpi/2021.9.0/include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skyla
            ke_avx512/numactl/2.0.14-gcc-zaa36r4/include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/netcdf-c/4.6.3-gcc-aq23kth/include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/hdf5/1.10.9-gcc-hbsptk3/inclu
            de -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/parmetis/4.0.3-gcc-nypuwzn/include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/metis/5.1.0-gcc-kougmmh/include -isystem /hppfs/work/pn49ha/di73yeq4/us
            er_spack23.1/linux-sles15-skylake_avx512/asagi/1.0.1-gcc-12.2.0-mqgdhwx/include -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/submodules/async -isystem /hppfs/work/pn
            49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/yaml-cpp/0.6.2-gcc-12.2.0-ak2irdf/lib/cmake/yaml-cpp/../../../include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/eigen/3.4.0-gcc-lbmyl47/include/eigen3 -isystem
             /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/src -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/submo
            dules -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/submodules/yateto/include -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y
            33ftnxsiyueorce3uobl/spack-build-r32ac7j/src -isystem /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/easi/1.3.0-gcc-12.2.0-to3va2m/include -O3 -DNDEBUG -std=c++17 -march=skylake-avx512 -I/hppfs/work/pn49ha/di73yeq4/
            user_spack23.1/linux-sles15-skylake_avx512/asagi/1.0.1-gcc-12.2.0-mqgdhwx/include -I/dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/netcdf-c/4.6.3-gcc-aq23kth/include -fopenmp -MD -MT CMakeFiles/seissol-lib.dir/src/Initializer/Time
            Stepping/LtsWeights/WeightsModels.cpp.o -MF CMakeFiles/seissol-lib.dir/src/Initializer/TimeStepping/LtsWeights/WeightsModels.cpp.o.d -o CMakeFiles/seissol-lib.dir/src/Initializer/TimeStepping/LtsWeights/WeightsModels.cpp.o -c /hppfs/scratc
            h/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/src/Initializer/TimeStepping/LtsWeights/WeightsModels.cpp
     474    [ 91%] Building CXX object CMakeFiles/seissol-lib.dir/src/Geometry/NetcdfReader.cpp.o
     475    /dss/lrzsys/sys/spack/user/release/23.1.0/skylake_avx512/spack/lib/spack/env/gcc/g++ -DALIGNED_REAL_SIZE=4 -DALIGNMENT=64 -DCONVERGENCE_ORDER=4 -DEIGEN_NO_CUDA -DH5_BUILT_AS_DYNAMIC_LIB -DLOGLEVEL0=2 -DLOGLEVEL="" -DLOG_LEVEL=2 -DNUMBER_OF
            _RELAXATION_MECHANISMS=0 -DOMP -DOMPI_SKIP_MPICXX -DPARALLEL -DREAL_SIZE=4 -DUSE_ASAGI -DUSE_ELASTIC -DUSE_HDF -DUSE_METIS -DUSE_MPI -DUSE_NETCDF -DUSE_NUMA_AWARE_PINNING -DUSE_PARMETIS -DUSE_PLASTICITY_NB -DVECTORSIZE=64 -D_FILE_OFFSET_BI
            TS=64 -D_GNU_SOURCE -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -D_POSIX_C_SOURCE=200809L -Ddunavant -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-build-r32ac7j/src/generat
            ed_code -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/src/Equations/elastic -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33
            ftnxsiyueorce3uobl/spack-src/src/Initializer/BatchRecorders -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/intel-oneapi-mpi/2021.9.0-gcc-xizuusf/mpi/2021.9.0/include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skyla
            ke_avx512/numactl/2.0.14-gcc-zaa36r4/include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/netcdf-c/4.6.3-gcc-aq23kth/include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/hdf5/1.10.9-gcc-hbsptk3/inclu
            de -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/parmetis/4.0.3-gcc-nypuwzn/include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/metis/5.1.0-gcc-kougmmh/include -isystem /hppfs/work/pn49ha/di73yeq4/us
            er_spack23.1/linux-sles15-skylake_avx512/asagi/1.0.1-gcc-12.2.0-mqgdhwx/include -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/submodules/async -isystem /hppfs/work/pn
            49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/yaml-cpp/0.6.2-gcc-12.2.0-ak2irdf/lib/cmake/yaml-cpp/../../../include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/eigen/3.4.0-gcc-lbmyl47/include/eigen3 -isystem
             /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/src -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/submo
            dules -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/submodules/yateto/include -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y
            33ftnxsiyueorce3uobl/spack-build-r32ac7j/src -isystem /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/easi/1.3.0-gcc-12.2.0-to3va2m/include -O3 -DNDEBUG -std=c++17 -march=skylake-avx512 -I/hppfs/work/pn49ha/di73yeq4/
            user_spack23.1/linux-sles15-skylake_avx512/asagi/1.0.1-gcc-12.2.0-mqgdhwx/include -I/dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/netcdf-c/4.6.3-gcc-aq23kth/include -fopenmp -MD -MT CMakeFiles/seissol-lib.dir/src/Geometry/NetcdfR
            eader.cpp.o -MF CMakeFiles/seissol-lib.dir/src/Geometry/NetcdfReader.cpp.o.d -o CMakeFiles/seissol-lib.dir/src/Geometry/NetcdfReader.cpp.o -c /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uo
            bl/spack-src/src/Geometry/NetcdfReader.cpp
     476    [ 92%] Building CXX object CMakeFiles/seissol-lib.dir/src/Geometry/CubeGenerator.cpp.o
     477    /dss/lrzsys/sys/spack/user/release/23.1.0/skylake_avx512/spack/lib/spack/env/gcc/g++ -DALIGNED_REAL_SIZE=4 -DALIGNMENT=64 -DCONVERGENCE_ORDER=4 -DEIGEN_NO_CUDA -DH5_BUILT_AS_DYNAMIC_LIB -DLOGLEVEL0=2 -DLOGLEVEL="" -DLOG_LEVEL=2 -DNUMBER_OF
            _RELAXATION_MECHANISMS=0 -DOMP -DOMPI_SKIP_MPICXX -DPARALLEL -DREAL_SIZE=4 -DUSE_ASAGI -DUSE_ELASTIC -DUSE_HDF -DUSE_METIS -DUSE_MPI -DUSE_NETCDF -DUSE_NUMA_AWARE_PINNING -DUSE_PARMETIS -DUSE_PLASTICITY_NB -DVECTORSIZE=64 -D_FILE_OFFSET_BI
            TS=64 -D_GNU_SOURCE -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -D_POSIX_C_SOURCE=200809L -Ddunavant -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-build-r32ac7j/src/generat
            ed_code -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/src/Equations/elastic -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33
            ftnxsiyueorce3uobl/spack-src/src/Initializer/BatchRecorders -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/intel-oneapi-mpi/2021.9.0-gcc-xizuusf/mpi/2021.9.0/include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skyla
            ke_avx512/numactl/2.0.14-gcc-zaa36r4/include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/netcdf-c/4.6.3-gcc-aq23kth/include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/hdf5/1.10.9-gcc-hbsptk3/inclu
            de -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/parmetis/4.0.3-gcc-nypuwzn/include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/metis/5.1.0-gcc-kougmmh/include -isystem /hppfs/work/pn49ha/di73yeq4/us
            er_spack23.1/linux-sles15-skylake_avx512/asagi/1.0.1-gcc-12.2.0-mqgdhwx/include -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/submodules/async -isystem /hppfs/work/pn
            49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/yaml-cpp/0.6.2-gcc-12.2.0-ak2irdf/lib/cmake/yaml-cpp/../../../include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/eigen/3.4.0-gcc-lbmyl47/include/eigen3 -isystem
             /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/src -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/submo
            dules -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/submodules/yateto/include -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y
            33ftnxsiyueorce3uobl/spack-build-r32ac7j/src -isystem /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/easi/1.3.0-gcc-12.2.0-to3va2m/include -O3 -DNDEBUG -std=c++17 -march=skylake-avx512 -I/hppfs/work/pn49ha/di73yeq4/
            user_spack23.1/linux-sles15-skylake_avx512/asagi/1.0.1-gcc-12.2.0-mqgdhwx/include -I/dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/netcdf-c/4.6.3-gcc-aq23kth/include -fopenmp -MD -MT CMakeFiles/seissol-lib.dir/src/Geometry/CubeGen
            erator.cpp.o -MF CMakeFiles/seissol-lib.dir/src/Geometry/CubeGenerator.cpp.o.d -o CMakeFiles/seissol-lib.dir/src/Geometry/CubeGenerator.cpp.o -c /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce
            3uobl/spack-src/src/Geometry/CubeGenerator.cpp
     478    /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/src/Reader/AsagiReader.cpp: In static member function 'static seissol::asagi::NumaCacheMode seissol::asagi::AsagiReader::getNumaM
            ode()':
  >> 479    /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-src/src/Reader/AsagiReader.cpp:105:45: error: cannot convert 'std::enable_if_t<true, std::__cxx11::basic_string<char> >' {aka 'std::_
            _cxx11::basic_string<char>'} to 'const char*' in initialization
     480      105 |   const char* numaModeName = utils::Env::get("SEISSOL_ASAGI_NUMA_MODE", "ON");
     481          |                              ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     482          |                                             |
     483          |                                             std::enable_if_t<true, std::__cxx11::basic_string<char> > {aka std::__cxx11::basic_string<char>}
  >> 484    make[2]: *** [CMakeFiles/seissol-lib.dir/build.make:723: CMakeFiles/seissol-lib.dir/src/Reader/AsagiReader.cpp.o] Error 1
     485    make[2]: *** Waiting for unfinished jobs....
     486    make[2]: Leaving directory '/hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-build-r32ac7j'
  >> 487    make[1]: *** [CMakeFiles/Makefile2:257: CMakeFiles/seissol-lib.dir/all] Error 2
     488    make[1]: Leaving directory '/hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-build-r32ac7j'
  >> 489    make: *** [Makefile:139: all] Error 2

See build log for details:
  /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-r32ac7jiux4y33ftnxsiyueorce3uobl/spack-build-out.txt
```
Some more info on the build:
```
di73yeq4@login04:~> spack spec -I seissol@mw2 convergence_order=4 dr_quad_rule=dunavant equations=elastic precision=single ^easi +python
Input spec
--------------------------------
 -   seissol@mw2 convergence_order=4 dr_quad_rule=dunavant equations=elastic precision=single
 -       ^easi+python

Concretized
--------------------------------
 -   seissol@mw2%gcc@12.2.0+asagi~cuda~intel_gpu~ipo+memkind+netcdf~rocm build_system=cmake build_type=Release convergence_order=4 dr_quad_rule=dunavant equations=elastic gemm_tools_list=LIBXSMM,PSpaMM generator=make graph_partitioning_libs=parmetis number_of_mechanisms=3 plasticity_method=nb precision=single arch=linux-sles15-skylake_avx512
[+]      ^asagi@1.0.1%gcc@12.2.0~examples+fortran~ipo+mpi+mpi3+numa~tests+threadsafe~threadsafe_counter+threadsafe_mpi build_system=cmake build_type=Release generator=make link_type=shared max_dimensions=4 arch=linux-sles15-skylake_avx512
[+]      ^cmake@3.27.7%gcc@12.2.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-sles15-skylake_avx512
[+]          ^curl@8.4.0%oneapi@2023.1.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-sles15-skylake_avx512
[+]              ^nghttp2@1.57.0%oneapi@2023.1.0 build_system=autotools arch=linux-sles15-skylake_avx512
[+]              ^openssl@3.1.3%oneapi@2023.1.0~docs+shared build_system=generic certs=mozilla arch=linux-sles15-skylake_avx512
[+]                  ^ca-certificates-mozilla@2023-05-30%oneapi@2023.1.0 build_system=generic arch=linux-sles15-skylake_avx512
[+]                  ^perl@5.38.0%oneapi@2023.1.0+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-sles15-skylake_avx512
[+]                      ^berkeley-db@18.1.40%oneapi@2023.1.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-sles15-skylake_avx512
[+]                      ^bzip2@1.0.8%oneapi@2023.1.0~debug~pic+shared build_system=generic arch=linux-sles15-skylake_avx512
[+]                          ^diffutils@3.9%oneapi@2023.1.0 build_system=autotools arch=linux-sles15-skylake_avx512
[+]                              ^libiconv@1.17%oneapi@2023.1.0 build_system=autotools libs=shared,static arch=linux-sles15-skylake_avx512
[+]                      ^gdbm@1.23%oneapi@2023.1.0 build_system=autotools arch=linux-sles15-skylake_avx512
[+]                          ^readline@8.2%oneapi@2023.1.0 build_system=autotools patches=bbf97f1 arch=linux-sles15-skylake_avx512
[+]              ^pkgconf@1.9.5%oneapi@2023.1.0 build_system=autotools arch=linux-sles15-skylake_avx512
[+]          ^ncurses@6.4%oneapi@2023.1.0~symlinks+termlib abi=none build_system=autotools arch=linux-sles15-skylake_avx512
[+]          ^zlib-ng@2.1.4%oneapi@2023.1.0+compat+opt build_system=autotools arch=linux-sles15-skylake_avx512
[+]      ^easi@1.3.0%gcc@12.2.0+asagi~ipo+python build_system=cmake build_type=Release generator=make jit=impalajit,lua arch=linux-sles15-skylake_avx512
[+]          ^impalajit@main%gcc@12.2.0~ipo+shared build_system=cmake build_type=Release generator=make arch=linux-sles15-skylake_avx512
[+]          ^lua@5.3.2%gcc@12.2.0~pcfile+shared build_system=makefile fetcher=curl arch=linux-sles15-skylake_avx512
[+]          ^py-pybind11@2.11.1%gcc@12.2.0~ipo build_system=cmake build_type=Release generator=ninja arch=linux-sles15-skylake_avx512
[^]      ^eigen@3.4.0%gcc@12.2.0~ipo build_system=cmake build_type=RelWithDebInfo generator=make arch=linux-sles15-skylake_avx512
[^]          ^cmake@3.26.3%gcc@12.2.0~doc+ncurses+ownlibs~qt build_system=generic build_type=Release arch=linux-sles15-skylake_avx512
[^]              ^ncurses@6.4%gcc@12.2.0~symlinks+termlib abi=none build_system=autotools arch=linux-sles15-skylake_avx512
[^]              ^openssl@1.1.1t%gcc@12.2.0~docs~shared build_system=generic certs=mozilla arch=linux-sles15-skylake_avx512
[^]                  ^ca-certificates-mozilla@2023-01-10%gcc@12.2.0 build_system=generic arch=linux-sles15-skylake_avx512
[^]                  ^perl@5.36.0%gcc@12.2.0+cpanm+open+shared+threads build_system=generic arch=linux-sles15-skylake_avx512
[^]                      ^gdbm@1.23%gcc@12.2.0 build_system=autotools arch=linux-sles15-skylake_avx512
[^]                          ^readline@8.2%gcc@12.2.0 build_system=autotools patches=bbf97f1 arch=linux-sles15-skylake_avx512
[^]          ^gmake@4.4.1%gcc@12.2.0~guile build_system=autotools arch=linux-sles15-skylake_avx512
[^]      ^gmake@4.4.1%gcc@12.2.0~guile build_system=generic arch=linux-sles15-skylake_avx512
[^]      ^hdf5@1.10.9%gcc@12.2.0+cxx+fortran+hl~ipo~java+mpi+shared+szip+threadsafe+tools api=default build_system=cmake build_type=Release generator=make arch=linux-sles15-skylake_avx512
[^]          ^cmake@3.26.3%gcc@12.2.0~doc+ncurses+ownlibs~qt build_system=generic build_type=Release arch=linux-sles15-skylake_avx512
[^]              ^openssl@1.1.1t%gcc@12.2.0~docs~shared build_system=generic certs=mozilla arch=linux-sles15-skylake_avx512
[^]          ^libaec@1.0.6%gcc@12.2.0~ipo+shared build_system=cmake build_type=Release generator=make arch=linux-sles15-skylake_avx512
[^]          ^pkgconf@1.8.0%gcc@12.2.0 build_system=autotools arch=linux-sles15-skylake_avx512
[^]          ^zlib@1.2.13%gcc@12.2.0+optimize+pic+shared build_system=makefile arch=linux-sles15-skylake_avx512
[^]      ^intel-oneapi-mpi@2021.9.0%gcc@12.2.0+envmods~external-libfabric~generic-names~ilp64 build_system=generic arch=linux-sles15-skylake_avx512
[^]      ^libxsmm@1.17%gcc@12.2.0~debug+generator+header-only~large_jit_buffer+shared blas=default build_system=makefile arch=linux-sles15-skylake_avx512
[^]          ^binutils@2.40%gcc@12.2.0+gas~gold~gprofng~headers~interwork+ld~libiberty~lto~nls~pgo+plugins build_system=autotools libs=shared,static arch=linux-sles15-skylake_avx512
[^]              ^diffutils@3.9%gcc@12.2.0 build_system=autotools arch=linux-sles15-skylake_avx512
[^]                  ^libiconv@1.17%gcc@12.2.0 build_system=autotools libs=shared,static arch=linux-sles15-skylake_avx512
[^]              ^texinfo@7.0%gcc@12.2.0 build_system=autotools arch=linux-sles15-skylake_avx512
[^]                  ^perl@5.36.0%gcc@12.2.0+cpanm+open+shared+threads build_system=generic arch=linux-sles15-skylake_avx512
[^]              ^zstd@1.5.5%gcc@12.2.0+programs build_system=makefile compression=none libs=shared,static arch=linux-sles15-skylake_avx512
[^]          ^python@3.10.10%gcc@12.2.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,7d40923,f2fd060 arch=linux-sles15-skylake_avx512
[^]              ^bzip2@1.0.8%gcc@12.2.0~debug~pic+shared build_system=generic arch=linux-sles15-skylake_avx512
[^]              ^expat@2.5.0%gcc@12.2.0+libbsd build_system=autotools arch=linux-sles15-skylake_avx512
[^]                  ^libbsd@0.11.7%gcc@12.2.0 build_system=autotools arch=linux-sles15-skylake_avx512
[^]                      ^libmd@1.0.4%gcc@12.2.0 build_system=autotools arch=linux-sles15-skylake_avx512
[^]              ^gdbm@1.23%gcc@12.2.0 build_system=autotools arch=linux-sles15-skylake_avx512
[^]              ^gettext@0.21.1%gcc@12.2.0+bzip2+curses+git~libunistring+libxml2+tar+xz build_system=autotools arch=linux-sles15-skylake_avx512
[^]                  ^libxml2@2.10.3%gcc@12.2.0~python build_system=autotools arch=linux-sles15-skylake_avx512
[^]                  ^tar@1.34%gcc@12.2.0 build_system=autotools zip=pigz arch=linux-sles15-skylake_avx512
[^]                      ^pigz@2.7%gcc@12.2.0 build_system=makefile arch=linux-sles15-skylake_avx512
[^]              ^libffi@3.4.4%gcc@12.2.0 build_system=autotools arch=linux-sles15-skylake_avx512
[^]              ^libxcrypt@4.4.33%gcc@12.2.0~obsolete_api build_system=autotools arch=linux-sles15-skylake_avx512
[^]              ^openssl@1.1.1t%gcc@12.2.0~docs~shared build_system=generic certs=mozilla arch=linux-sles15-skylake_avx512
[^]              ^readline@8.2%gcc@12.2.0 build_system=autotools patches=bbf97f1 arch=linux-sles15-skylake_avx512
[^]              ^sqlite@3.40.1%gcc@12.2.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-sles15-skylake_avx512
[^]              ^util-linux-uuid@2.38.1%gcc@12.2.0 build_system=autotools arch=linux-sles15-skylake_avx512
[^]              ^xz@5.4.1%gcc@12.2.0~pic build_system=autotools libs=shared,static arch=linux-sles15-skylake_avx512
[^]      ^memkind@1.13.0%gcc@12.2.0 build_system=autotools arch=linux-sles15-skylake_avx512
[^]          ^autoconf@2.71%gcc@12.2.0 build_system=autotools arch=linux-sles15-skylake_avx512
[^]              ^perl@5.36.0%gcc@12.2.0+cpanm+open+shared+threads build_system=generic arch=linux-sles15-skylake_avx512
[^]                  ^berkeley-db@18.1.40%gcc@12.2.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-sles15-skylake_avx512
[^]          ^automake@1.16.5%gcc@12.2.0 build_system=autotools arch=linux-sles15-skylake_avx512
[^]          ^libtool@2.4.7%gcc@12.2.0 build_system=autotools arch=linux-sles15-skylake_avx512
[^]          ^m4@1.4.19%gcc@12.2.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-sles15-skylake_avx512
[^]              ^libsigsegv@2.14%gcc@12.2.0 build_system=autotools arch=linux-sles15-skylake_avx512
[^]          ^numactl@2.0.14%gcc@12.2.0 build_system=autotools patches=4e1d78c,62fc8a8,ff37630 arch=linux-sles15-skylake_avx512
[^]      ^metis@5.1.0%gcc@12.2.0~gdb+int64~ipo~real64+shared build_system=cmake build_type=Release generator=make patches=4991da9,93a7903,b1225da arch=linux-sles15-skylake_avx512
[^]      ^netcdf-c@4.6.3%gcc@12.2.0~blosc~byterange~dap~fsync~hdf4~jna+mpi~nczarr_zip+optimize+parallel-netcdf+pic+shared~szip~zstd build_system=autotools arch=linux-sles15-skylake_avx512
[^]          ^parallel-netcdf@1.11.2%gcc@12.2.0~burstbuffer+cxx+fortran+pic+shared build_system=autotools arch=linux-sles15-skylake_avx512
[^]      ^parmetis@4.0.3%gcc@12.2.0~gdb+int64~ipo+shared build_system=cmake build_type=Release generator=make patches=4f89253,50ed208,704b84f arch=linux-sles15-skylake_avx512
[+]      ^py-pspamm@develop%gcc@12.2.0~numpy~scipy build_system=python_pip arch=linux-sles15-skylake_avx512
[+]          ^py-pip@23.1.2%gcc@12.2.0 build_system=generic arch=linux-sles15-skylake_avx512
[+]          ^py-wheel@0.41.2%gcc@12.2.0 build_system=generic arch=linux-sles15-skylake_avx512
[+]      ^py-setuptools@68.0.0%gcc@12.2.0 build_system=generic arch=linux-sles15-skylake_avx512
[e]      ^python@3.10.10%gcc@12.2.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,7d40923,ebdca64,f2fd060 arch=linux-sles15-skylake_avx512
[+]      ^yaml-cpp@0.6.2%gcc@12.2.0~ipo+pic+shared~tests build_system=cmake build_type=Release generator=make arch=linux-sles15-skylake_avx512
```

same issue with intel actually:
(icpc (ICC) 2021.9.0 20230302)
```
==> Installing seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx [42/42]
==> No binary for seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx found: installing from source
==> No patches needed for seissol
==> seissol: Executing phase: 'cmake'
==> seissol: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j40'

4 errors found in build log:
     773    icpc: remark #10441: The Intel(R) C++ Compiler Classic (ICC) is deprecated and will be removed from product release in the second half of 2023. The Intel(R) oneAPI DPC++/C++ Compiler (ICX) is the recommended compiler moving forward. Please
             transition to use this compiler. Use '-diag-disable=10441' to disable this message.
     774    icpc: command line warning #10121: overriding '-march=skylake-avx512' with '-xCORE-AVX512'
     775    icpc: remark #10441: The Intel(R) C++ Compiler Classic (ICC) is deprecated and will be removed from product release in the second half of 2023. The Intel(R) oneAPI DPC++/C++ Compiler (ICX) is the recommended compiler moving forward. Please
             transition to use this compiler. Use '-diag-disable=10441' to disable this message.
     776    icpc: command line warning #10121: overriding '-march=skylake-avx512' with '-xCORE-AVX512'
     777    [ 90%] Building CXX object CMakeFiles/seissol-lib.dir/src/Initializer/TimeStepping/LtsWeights/WeightsModels.cpp.o
     778    /dss/lrzsys/sys/spack/user/release/23.1.0/skylake_avx512/spack/lib/spack/env/intel/icpc -DALIGNED_REAL_SIZE=4 -DALIGNMENT=64 -DCONVERGENCE_ORDER=4 -DEIGEN_NO_CUDA -DEXPERIMENTAL_FS -DH5_BUILT_AS_DYNAMIC_LIB -DLOGLEVEL0=2 -DLOGLEVEL="" -DLO
            G_LEVEL=2 -DNUMBER_OF_RELAXATION_MECHANISMS=0 -DOMP -DOMPI_SKIP_MPICXX -DPARALLEL -DREAL_SIZE=4 -DUSE_ASAGI -DUSE_ELASTIC -DUSE_HDF -DUSE_METIS -DUSE_MPI -DUSE_NETCDF -DUSE_NUMA_AWARE_PINNING -DUSE_PARMETIS -DUSE_PLASTICITY_NB -DVECTORSIZE
            =64 -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -D_POSIX_C_SOURCE=200809L -Ddunavant -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-buil
            d-tnshtsr/src/generated_code -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-src/src/Equations/elastic -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seiss
            ol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-src/src/Initializer/BatchRecorders -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/intel-oneapi-mpi/2021.9.0-intel-rcnbyxw/mpi/2021.9.0/include -isystem /dss/lrzsys/sys/spack/r
            elease/23.1.0/opt/skylake_avx512/numactl/2.0.14-intel-drj2eve/include -isystem /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/netcdf-c/4.8.1-intel-2023.1.0-bgfnea2/include -isystem /dss/lrzsys/sys/spack/release/23.1
            .0/opt/skylake_avx512/hdf5/1.10.9-intel-epoi76h/include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/parmetis/4.0.3-intel-53ujuh2/include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/metis/5.1.0-inte
            l-2munz63/include -isystem /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/asagi/1.0.1-intel-2023.1.0-duitc4d/include -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtx
            n5fraabgyx/spack-src/submodules/async -isystem /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/yaml-cpp/0.6.2-intel-2023.1.0-sydwyux/lib/cmake/yaml-cpp/../../../include -isystem /dss/lrzsys/sys/spack/release/23.1.0/o
            pt/skylake_avx512/eigen/3.4.0-intel-vfictvg/include/eigen3 -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-src/src -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack
            -stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-src/submodules -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-src/submodules/yateto/include -isystem /hppfs/scrat
            ch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-build-tnshtsr/src -isystem /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/easi/1.3.0-intel-2023.1.0-sv2vrge/include -std=
            c++17 -O3 -DNDEBUG -std=c++17 -xCORE-AVX512 -fma -I/hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/asagi/1.0.1-intel-2023.1.0-duitc4d/include -I/hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/n
            etcdf-c/4.8.1-intel-2023.1.0-bgfnea2/include -qopenmp -MD -MT CMakeFiles/seissol-lib.dir/src/Initializer/TimeStepping/LtsWeights/WeightsModels.cpp.o -MF CMakeFiles/seissol-lib.dir/src/Initializer/TimeStepping/LtsWeights/WeightsModels.cpp.o
            .d -o CMakeFiles/seissol-lib.dir/src/Initializer/TimeStepping/LtsWeights/WeightsModels.cpp.o -c /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-src/src/Initializer/TimeStepping/LtsW
            eights/WeightsModels.cpp
  >> 779    /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-src/src/Reader/AsagiReader.cpp(105): error: no suitable conversion function from "std::enable_if_t<true, std::__cxx11::string>" to "c
            onst char *" exists
     780        const char* numaModeName = utils::Env::get("SEISSOL_ASAGI_NUMA_MODE", "ON");
     781                                   ^
     782    
     783    icpc: remark #10441: The Intel(R) C++ Compiler Classic (ICC) is deprecated and will be removed from product release in the second half of 2023. The Intel(R) oneAPI DPC++/C++ Compiler (ICX) is the recommended compiler moving forward. Please
             transition to use this compiler. Use '-diag-disable=10441' to disable this message.
     784    icpc: command line warning #10121: overriding '-march=skylake-avx512' with '-xCORE-AVX512'
     785    compilation aborted for /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-src/src/Reader/AsagiReader.cpp (code 2)
  >> 786    make[2]: *** [CMakeFiles/seissol-lib.dir/build.make:723: CMakeFiles/seissol-lib.dir/src/Reader/AsagiReader.cpp.o] Error 2
     787    make[2]: *** Waiting for unfinished jobs....
     788    [ 91%] Building CXX object CMakeFiles/seissol-lib.dir/src/Geometry/NetcdfReader.cpp.o
     789    /dss/lrzsys/sys/spack/user/release/23.1.0/skylake_avx512/spack/lib/spack/env/intel/icpc -DALIGNED_REAL_SIZE=4 -DALIGNMENT=64 -DCONVERGENCE_ORDER=4 -DEIGEN_NO_CUDA -DEXPERIMENTAL_FS -DH5_BUILT_AS_DYNAMIC_LIB -DLOGLEVEL0=2 -DLOGLEVEL="" -DLO
            G_LEVEL=2 -DNUMBER_OF_RELAXATION_MECHANISMS=0 -DOMP -DOMPI_SKIP_MPICXX -DPARALLEL -DREAL_SIZE=4 -DUSE_ASAGI -DUSE_ELASTIC -DUSE_HDF -DUSE_METIS -DUSE_MPI -DUSE_NETCDF -DUSE_NUMA_AWARE_PINNING -DUSE_PARMETIS -DUSE_PLASTICITY_NB -DVECTORSIZE
            =64 -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -D_POSIX_C_SOURCE=200809L -Ddunavant -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-buil
            d-tnshtsr/src/generated_code -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-src/src/Equations/elastic -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seiss
            ol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-src/src/Initializer/BatchRecorders -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/intel-oneapi-mpi/2021.9.0-intel-rcnbyxw/mpi/2021.9.0/include -isystem /dss/lrzsys/sys/spack/r
            elease/23.1.0/opt/skylake_avx512/numactl/2.0.14-intel-drj2eve/include -isystem /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/netcdf-c/4.8.1-intel-2023.1.0-bgfnea2/include -isystem /dss/lrzsys/sys/spack/release/23.1
            .0/opt/skylake_avx512/hdf5/1.10.9-intel-epoi76h/include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/parmetis/4.0.3-intel-53ujuh2/include -isystem /dss/lrzsys/sys/spack/release/23.1.0/opt/skylake_avx512/metis/5.1.0-inte
            l-2munz63/include -isystem /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/asagi/1.0.1-intel-2023.1.0-duitc4d/include -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtx
            n5fraabgyx/spack-src/submodules/async -isystem /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/yaml-cpp/0.6.2-intel-2023.1.0-sydwyux/lib/cmake/yaml-cpp/../../../include -isystem /dss/lrzsys/sys/spack/release/23.1.0/o
            pt/skylake_avx512/eigen/3.4.0-intel-vfictvg/include/eigen3 -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-src/src -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack
            -stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-src/submodules -isystem /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-src/submodules/yateto/include -isystem /hppfs/scrat
            ch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-build-tnshtsr/src -isystem /hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/easi/1.3.0-intel-2023.1.0-sv2vrge/include -std=
            c++17 -O3 -DNDEBUG -std=c++17 -xCORE-AVX512 -fma -I/hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/asagi/1.0.1-intel-2023.1.0-duitc4d/include -I/hppfs/work/pn49ha/di73yeq4/user_spack23.1/linux-sles15-skylake_avx512/n
            etcdf-c/4.8.1-intel-2023.1.0-bgfnea2/include -qopenmp -MD -MT CMakeFiles/seissol-lib.dir/src/Geometry/NetcdfReader.cpp.o -MF CMakeFiles/seissol-lib.dir/src/Geometry/NetcdfReader.cpp.o.d -o CMakeFiles/seissol-lib.dir/src/Geometry/NetcdfRead
            er.cpp.o -c /hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-src/src/Geometry/NetcdfReader.cpp
     790    icpc: remark #10441: The Intel(R) C++ Compiler Classic (ICC) is deprecated and will be removed from product release in the second half of 2023. The Intel(R) oneAPI DPC++/C++ Compiler (ICX) is the recommended compiler moving forward. Please
             transition to use this compiler. Use '-diag-disable=10441' to disable this message.
     791    icpc: command line warning #10121: overriding '-march=skylake-avx512' with '-xCORE-AVX512'
     792    make[2]: Leaving directory '/hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-build-tnshtsr'
  >> 793    make[1]: *** [CMakeFiles/Makefile2:257: CMakeFiles/seissol-lib.dir/all] Error 2
     794    make[1]: Leaving directory '/hppfs/scratch/0A/di73yeq4/tmp/build_stage/spack-stage-seissol-mw2-tnshtsrsq5qbnduufjkdtxn5fraabgyx/spack-build-tnshtsr'
  >> 795    make: *** [Makefile:139: all] Error 2
```

--- a/.github/workflows/build-seissol-cpu.yml
+++ b/.github/workflows/build-seissol-cpu.yml
@@ -87,7 +87,7 @@ jobs:
             TESTING_COMMAND="";
           fi;
 
-          cmake .. -GNinja -DTESTING_COMMAND="${TESTING_COMMAND}" -DCMAKE_INSTALL_PREFIX=/opt/seissol -DTESTING=ON -DHOST_ARCH=${{matrix.arch}} -DORDER=${{matrix.order}} -DCMAKE_BUILD_TYPE=${{matrix.build_type}} -DEQUATIONS=${{matrix.equation}} -DPRECISION=${{matrix.precision}} -DNUMBER_OF_MECHANISMS=$mechanisms
+          cmake .. -GNinja -DTESTING_COMMAND="${TESTING_COMMAND}" -DCMAKE_INSTALL_PREFIX=/opt/seissol -DTESTING=ON -DASAGI=ON -DHOST_ARCH=${{matrix.arch}} -DORDER=${{matrix.order}} -DCMAKE_BUILD_TYPE=${{matrix.build_type}} -DEQUATIONS=${{matrix.equation}} -DPRECISION=${{matrix.precision}} -DNUMBER_OF_MECHANISMS=$mechanisms
           ninja install
 
       - name: test-seissol

--- a/.github/workflows/build-seissol-gpu.yml
+++ b/.github/workflows/build-seissol-gpu.yml
@@ -115,7 +115,7 @@ jobs:
           export CXX=${{matrix.setup.cxx}}
           export FC=${{matrix.setup.fc}}
 
-          cmake .. -GNinja -DTESTING=OFF -DDEVICE_BACKEND=${{matrix.setup.backend}} -DDEVICE_ARCH=${{matrix.setup.arch}} -DSYCL_CC=${{matrix.setup.syclcc}} -DHOST_ARCH=${{matrix.arch}} -DORDER=${{matrix.order}} -DCMAKE_BUILD_TYPE=${{matrix.build_type}} -DEQUATIONS=${{matrix.equation}} -DPRECISION=${{matrix.precision}} -DNUMBER_OF_MECHANISMS=$mechanisms -DCMAKE_INSTALL_PREFIX=/opt/seissol
+          cmake .. -GNinja -DTESTING=OFF -DDEVICE_BACKEND=${{matrix.setup.backend}} -DDEVICE_ARCH=${{matrix.setup.arch}} -DSYCL_CC=${{matrix.setup.syclcc}} -DHOST_ARCH=${{matrix.arch}} -DORDER=${{matrix.order}} -DCMAKE_BUILD_TYPE=${{matrix.build_type}} -DEQUATIONS=${{matrix.equation}} -DPRECISION=${{matrix.precision}} -DNUMBER_OF_MECHANISMS=$mechanisms -DASAGI=ON -DCMAKE_INSTALL_PREFIX=/opt/seissol
           ninja install
 
       - name: upload-seissol

--- a/src/Reader/AsagiModule.cpp
+++ b/src/Reader/AsagiModule.cpp
@@ -36,13 +36,16 @@ AsagiModule::AsagiModule() : m_mpiMode(getMPIMode()), m_totalThreads(getTotalThr
 
 AsagiMPIMode AsagiModule::getMPIMode() {
 #ifdef USE_MPI
-  std::string mpiModeName = utils::Env::get(EnvMPIMode, "WINDOWS");
-  if (mpiModeName == "WINDOWS")
+  const std::string mpiModeName = utils::Env::get(EnvMPIMode, "WINDOWS");
+  if (mpiModeName == "WINDOWS") {
     return AsagiMPIMode::Windows;
-  if (mpiModeName == "COMM_THREAD")
+  }
+  if (mpiModeName == "COMM_THREAD") {
     return AsagiMPIMode::CommThread;
-  if (mpiModeName == "OFF")
+  }
+  if (mpiModeName == "OFF") {
     return AsagiMPIMode::Off;
+  }
 
   return AsagiMPIMode::Unknown;
 #else  // USE_MPI
@@ -75,7 +78,7 @@ void AsagiModule::preMPI() {
 
 void AsagiModule::postMPIInit() {
   if (m_mpiMode == AsagiMPIMode::Unknown) {
-    std::string mpiModeName = utils::Env::get(EnvMPIMode, "");
+    const std::string mpiModeName = utils::Env::get(EnvMPIMode, "");
     logError() << "Unknown ASAGI MPI mode:" << mpiModeName;
   } else {
     logWarning() << "Running with only one OMP thread."

--- a/src/Reader/AsagiModule.cpp
+++ b/src/Reader/AsagiModule.cpp
@@ -11,7 +11,11 @@
 #include "AsagiModule.h"
 #include "Parallel/Helper.h"
 #include "utils/env.h"
+#include <Modules/Modules.h>
+#include <Parallel/MPI.h>
+#include <asagi.h>
 #include <string>
+#include <utils/logger.h>
 
 #ifdef _OPENMP
 #include <omp.h>
@@ -36,7 +40,7 @@ AsagiModule::AsagiModule() : m_mpiMode(getMPIMode()), m_totalThreads(getTotalThr
 
 AsagiMPIMode AsagiModule::getMPIMode() {
 #ifdef USE_MPI
-  const std::string mpiModeName = utils::Env::get(EnvMPIMode, "WINDOWS");
+  const std::string mpiModeName = utils::Env::get(EnvMpiMode, "WINDOWS");
   if (mpiModeName == "WINDOWS") {
     return AsagiMPIMode::Windows;
   }
@@ -78,7 +82,7 @@ void AsagiModule::preMPI() {
 
 void AsagiModule::postMPIInit() {
   if (m_mpiMode == AsagiMPIMode::Unknown) {
-    const std::string mpiModeName = utils::Env::get(EnvMPIMode, "");
+    const std::string mpiModeName = utils::Env::get(EnvMpiMode, "");
     logError() << "Unknown ASAGI MPI mode:" << mpiModeName;
   } else {
     logWarning() << "Running with only one OMP thread."

--- a/src/Reader/AsagiModule.h
+++ b/src/Reader/AsagiModule.h
@@ -11,17 +11,11 @@
 
 #ifdef USE_ASAGI
 
-#include "Parallel/MPI.h"
-
 #include <string>
 
 #include <asagi.h>
 
-#include "utils/env.h"
-#include "utils/logger.h"
-
 #include "Modules/Module.h"
-#include "Modules/Modules.h"
 
 namespace seissol::asagi {
 
@@ -40,8 +34,11 @@ class AsagiModule : public Module {
   AsagiModule();
 
   public:
+  ~AsagiModule() override = default;
   AsagiModule(const AsagiModule&) = delete;
   void operator=(const AsagiModule&) = delete;
+  AsagiModule(AsagiModule&&) = delete;
+  void operator=(AsagiModule&&) = delete;
 
   void preMPI() override;
 
@@ -90,7 +87,7 @@ class AsagiModule : public Module {
   static int totalThreads();
 
   private:
-  static inline const char* EnvMPIMode = "SEISSOL_ASAGI_MPI_MODE";
+  static inline const std::string EnvMpiMode = "SEISSOL_ASAGI_MPI_MODE";
 };
 
 } // namespace seissol::asagi

--- a/src/Reader/AsagiReader.cpp
+++ b/src/Reader/AsagiReader.cpp
@@ -9,6 +9,13 @@
 
 #include "AsagiReader.h"
 
+#include <Monitoring/Instrumentation.h>
+#include <Reader/AsagiModule.h>
+#include <asagi.h>
+#include <mpi.h>
+#include <utils/env.h>
+#include <utils/logger.h>
+
 namespace seissol::asagi {
 /**
  *
@@ -21,25 +28,27 @@ namespace seissol::asagi {
 
   ::asagi::Grid* grid = ::asagi::Grid::createArray();
 
-  if (utils::Env::get<bool>((envPrefix + "_SPARSE").c_str(), false)) {
+  if (utils::Env::get<bool>(envPrefix + "_SPARSE", false)) {
     grid->setParam("GRID", "CACHE");
   }
 
   // Set MPI mode
   if (AsagiModule::mpiMode() != AsagiMPIMode::Off) {
 #ifdef USE_MPI
-    ::asagi::Grid::Error err = grid->setComm(comm);
-    if (err != ::asagi::Grid::SUCCESS)
+    ::asagi::Grid::Error const err = grid->setComm(comm);
+    if (err != ::asagi::Grid::SUCCESS) {
       logError() << "Could not set ASAGI communicator:" << err;
+    }
 
 #endif // USE_MPI
 
-    if (AsagiModule::mpiMode() == AsagiMPIMode::CommThread)
+    if (AsagiModule::mpiMode() == AsagiMPIMode::CommThread) {
       grid->setParam("MPI_COMMUNICATION", "THREAD");
+    }
   }
 
   // Set NUMA mode
-  asagiThreads = utils::Env::get((envPrefix + "_NUM_THREADS").c_str(), 0u);
+  asagiThreads = utils::Env::get(envPrefix + "_NUM_THREADS", 0U);
   if (asagiThreads == 0) {
     asagiThreads = AsagiModule::totalThreads();
   } else if (static_cast<int>(asagiThreads) > AsagiModule::totalThreads()) {
@@ -71,12 +80,12 @@ namespace seissol::asagi {
   grid->setParam("VALUE_POSITION", "VERTEX_CENTERED");
 
   // Set additional parameters
-  const std::string blockSize = utils::Env::get((envPrefix + "_BLOCK_SIZE").c_str(), "64");
+  const std::string blockSize = utils::Env::get(envPrefix + "_BLOCK_SIZE", "64");
   grid->setParam("BLOCK_SIZE_0", blockSize.c_str());
   grid->setParam("BLOCK_SIZE_1", blockSize.c_str());
   grid->setParam("BLOCK_SIZE_2", blockSize.c_str());
 
-  const std::string cacheSize = utils::Env::get((envPrefix + "_CACHE_SIZE").c_str(), "128");
+  const std::string cacheSize = utils::Env::get(envPrefix + "_CACHE_SIZE", "128");
   grid->setParam("CACHE_SIZE", cacheSize.c_str());
 
   grid->setParam("VARIABLE", varname);
@@ -88,9 +97,10 @@ namespace seissol::asagi {
 #pragma omp parallel shared(abort) num_threads(asagiThreads)
 #endif // _OPENMP
   {
-    ::asagi::Grid::Error err = grid->open(file);
-    if (err != ::asagi::Grid::SUCCESS)
+    const ::asagi::Grid::Error err = grid->open(file);
+    if (err != ::asagi::Grid::SUCCESS) {
       abort = true;
+    }
   }
   // SCOREP_RECORDING_ON();
   if (abort) {

--- a/src/Reader/AsagiReader.cpp
+++ b/src/Reader/AsagiReader.cpp
@@ -71,12 +71,12 @@ namespace seissol::asagi {
   grid->setParam("VALUE_POSITION", "VERTEX_CENTERED");
 
   // Set additional parameters
-  std::string blockSize = utils::Env::get((envPrefix + "_BLOCK_SIZE").c_str(), "64");
+  const std::string blockSize = utils::Env::get((envPrefix + "_BLOCK_SIZE").c_str(), "64");
   grid->setParam("BLOCK_SIZE_0", blockSize.c_str());
   grid->setParam("BLOCK_SIZE_1", blockSize.c_str());
   grid->setParam("BLOCK_SIZE_2", blockSize.c_str());
 
-  std::string cacheSize = utils::Env::get((envPrefix + "_CACHE_SIZE").c_str(), "128");
+  const std::string cacheSize = utils::Env::get((envPrefix + "_CACHE_SIZE").c_str(), "128");
   grid->setParam("CACHE_SIZE", cacheSize.c_str());
 
   grid->setParam("VARIABLE", varname);
@@ -102,14 +102,17 @@ namespace seissol::asagi {
 }
 
 NumaCacheMode AsagiReader::getNumaMode() {
-  const char* numaModeName = utils::Env::get("SEISSOL_ASAGI_NUMA_MODE", "ON");
+  const std::string numaModeName = utils::Env::get("SEISSOL_ASAGI_NUMA_MODE", "ON");
 
-  if (strcmp(numaModeName, "ON") == 0)
+  if (numaModeName == "ON") {
     return NumaCacheMode::On;
-  if (strcmp(numaModeName, "OFF") == 0)
+  }
+  if (numaModeName == "OFF") {
     return NumaCacheMode::Off;
-  if (strcmp(numaModeName, "CACHE") == 0)
+  }
+  if (numaModeName == "CACHE") {
     return NumaCacheMode::Cache;
+  }
 
   logError() << "Unknown NUMA mode:" << numaModeName;
   return NumaCacheMode::Off;

--- a/src/Reader/AsagiReader.h
+++ b/src/Reader/AsagiReader.h
@@ -15,12 +15,6 @@
 #include "easi/util/AsagiReader.h"
 #include <asagi.h>
 
-#include "utils/env.h"
-#include "utils/logger.h"
-
-#include "AsagiModule.h"
-#include "Monitoring/Instrumentation.h"
-
 namespace seissol::asagi {
 enum class NumaCacheMode { Off, On, Cache };
 
@@ -45,8 +39,8 @@ class AsagiReader : public easi::AsagiReader {
 #endif
   );
 
-  virtual ::asagi::Grid* open(const char* file, const char* varname);
-  virtual unsigned numberOfThreads() const;
+  ::asagi::Grid* open(const char* file, const char* varname) override;
+  [[nodiscard]] unsigned numberOfThreads() const override;
 
   private:
   static NumaCacheMode getNumaMode();

